quiz-app-api/app/controllers/api/TeamController.js

```diff
@@ -1,83 +1,128 @@
+    if (req.body.data.password) {
+      req.body.data.password = hashPassword(req.body.data.password)
     }
-    bcrypt.hash(req.body.password, 10, (err, hash) => {
-      teamObj.password = hash
-      tc.update(teamObj, req.params._id).then((result) => {
-        res.status(200).json(result);
-      });
-    })
 
-  },
-  show: (req, res) => {
-    tc.show(req.params._id).then((result) => {
-      res.status(200).json(result);
-    });
+    // Remove leader's id from members array
+    req.body.data.members = req.body.data.members.filter(member => String(member) !== req.userId)
 
+    super.update(req, res)
+  }
 
-  },
+  addMember(req, res) {
+    // This method expects the username of the member to be added to the team
+    const id = req.params._id
+    User.findOne({ username: req.body.data.username })
+      .exec()
+      .then(user => {
+        if (user) {
+          // User with the given username exists, now add him to the team
+          console.log(user)
+          Team.update({_id: id}, { $addToSet: { members: db.mongo.ObjectID(user._id) } })
+            .exec()
+            .then(result => {
+              console.log(result)
+              res.status(200).json({
+                message: result.nModified + " member(s) added.",
+                count: result.nModified
+              })
+            })
+            .catch(error => {
+              console.log(error)
+              res.status(500).json({ error })
+            })
+        }
+      })
+      .catch(error => {
+        console.log(error);
+        res.status(500).json({ error });
+      })
+  }
 
+  removeMember(req, res) {
+    // This method won't allow to remove the team leader from the members
+    // This method expects the username of the member to be added to the team
+    const id = req.params._id
+    User.findOne({ username: req.body.data.username })
+      .exec()
+      .then(user => {
+        if (user && req.userId !== String(user._id)) {
+          // User with the given username exists, now add remove him from the team
+          Team.update({_id: id}, { $pull: { members: user._id } })
+            .exec()
+            .then(result => {
+              console.log(result)
+              res.status(200).json({
+                message: result.nModified + " member(s) removed.",
+                count: result.nModified
+              })
+            })
+            .catch(error => {
+              console.log(error)
+              res.status(500).json({ error })
+            })
+        } else {
+          res.status(400).json({
+            message: 'Cannot remove team leader'
+          })
+        }
+      })
+      .catch(error => {
+        console.log(error);
+        res.status(500).json({ error });
+      })
+  }
+
+  login(req, res) {
+    // TODO: Implement this later
+    // This will require to modify Team Model to add a field to uniquely indentify a team
+    // from user's perspective
+    // FIXME:
```

Comment:
We will need a unique identifier for every team so that they can log in. Something like a Team Code or Team Username. We can also leave this for later